collector/kernel/kernel_collector.cc

```diff
@@ -278,47 +278,53 @@ void KernelCollector::on_authenticated()
 
 void KernelCollector::probe_holdoff_timeout(uv_timer_t *timer)
 {
+  auto const upstream_connection_flush_and_close = [this] () {
+    upstream_connection_.flush();
+    upstream_connection_.close();
+  };
+
   if (entrypoint_error_ != EntrypointError::none) {
-    print_troubleshooting_message_and_exit(log_, host_info_, entrypoint_error_);
+    print_troubleshooting_message_and_exit(host_info_, entrypoint_error_, log_, upstream_connection_flush_and_close);
+    return;
   }
 
   LOG::trace("Adding probes");
 
   last_probe_monotonic_time_ns_ = monotonic();
 
+  auto const handle_exception = [&, this] (TroubleshootItem item, std::exception const &e) {
```

Comment:
Just curious, why capturing this is needed here?

Reply:
I originally was just capturing this, then I added some other items that needed to be captured so I added &, but & by itself should be sufficient.  I'll remove "this" from the capture block.

Reply:
Removed.